apps/pipeline-asset-sg/src/export-to-view.service.ts

```diff
@@ -0,0 +1,334 @@
+      await this.export('assetFormatComposition', 'assetId', assetIds);
+      await this.export('assetKindComposition', 'assetId', assetIds);
+      await this.exportInternalProjects(assetIds);
+      await this.export('assetLanguage', 'assetId', assetIds);
+      await this.exportPublications(assetIds);
+      await this.exportSiblings(assetIds, publicAssetIds);
+
+      await this.export('autoCat', 'assetId', assetIds);
+      await this.export('manCatLabelRef', 'assetId', assetIds);
+      await this.export('statusWork', 'assetId', assetIds);
+      await this.exportStudyAreas(assetIds);
+      await this.exportStudyLocations(assetIds);
+      await this.exportStudyTraces(assetIds);
+      await this.export('typeNatRel', 'assetId', assetIds);
+
+      const timeTaken = Date.now() - time;
+      log(`Exported batch of ${assetIds.length} assets in ${timeTaken} ms.`);
+    }
+
+    await this.exportFiles(publicAssetIds);
+  }
+
+  /**
+   * Export study areas as unsafe query with st_astext.
+   */
+  private async exportStudyAreas(assetIds: number[]) {
+    const studyAreas = await this.sourcePrisma.$queryRaw<
+      { study_area_id: number; asset_id: number; geom_quality_item_code: string; geom: string }[]
+    >`
+      SELECT study_area_id, asset_id, geom_quality_item_code, st_astext(geom, 2056) as geom
+      FROM study_area WHERE asset_id IN (${Prisma.join(assetIds)})
+    `;
+
+    const formattedStudyAreas = studyAreas.map(
+      (sa) =>
+        Prisma.sql`(${sa.study_area_id}, ${sa.asset_id}, ${sa.geom_quality_item_code}, ST_GeomFromText(${sa.geom}, 2056))`
+    );
+
+    // insert into destination as raw query
+    const result = await this.destinationPrisma.$executeRaw`
+      INSERT INTO study_area (study_area_id, asset_id, geom_quality_item_code, geom)
+      VALUES ${Prisma.join(formattedStudyAreas)}
+    `;
+
+    log(`Created ${result} study areas.`);
+  }
+
+  /**
+   * Export study locations as unsafe query with st_astext.
+   */
+  private async exportStudyLocations(assetIds: number[]) {
+    const studyLocations = await this.sourcePrisma.$queryRaw<
+      { study_location_id: number; asset_id: number; geom_quality_item_code: string; geom: string }[]
+    >`
+      SELECT study_location_id, asset_id, geom_quality_item_code, st_astext(geom, 2056) as geom
+      FROM study_location WHERE asset_id IN (${Prisma.join(assetIds)})
+    `;
+
+    const formattedStudyLocations = studyLocations.map(
+      (sa) =>
+        Prisma.sql`(${sa.study_location_id}, ${sa.asset_id}, ${sa.geom_quality_item_code}, ST_GeomFromText(${sa.geom}, 2056))`
+    );
+
+    // insert into destination as raw query
+    const result = await this.destinationPrisma.$executeRaw`
+      INSERT INTO study_location (study_location_id, asset_id, geom_quality_item_code, geom)
+      VALUES ${Prisma.join(formattedStudyLocations)}
+    `;
+
+    log(`Created ${result} study locations.`);
+  }
+
+  /**
+   * Export study traces as unsafe query with st_astext.
```

Comment:
Just a quick note that Prisma also has _unsafe_ raw queries, so this comment is a bit misleading, as in fact, we're using _safe_ raw queries. The safety part is just refering to whether the query is a simple, unchecked string (unsafe) or an instance of `Prisma.Sql` (safe).